### PR TITLE
completed get /api/articles

### PR DESCRIPTION
**Pull request by drewshore24**

### CODE REVIEW

**Comment by drewshore24** [comment]
realised I've label this with wrong question - correct question title for this was 
CORE: GET /api/articles/:article_id

File: __tests__/endpoint.test.js [hunk 1]
```diff
@@ -42,4 +42,40 @@ describe('Endpoint Tests', () => {
             })
         })
     })
+    describe('/api/articles/:article_id', () => {
+        test('200: get 200 response', () => {
+            return request(app).get('/api/articles/1').expect(200)
```
Comment (nc-john-hutcheon) [inline __tests__/endpoint.test.js:47]: This expect can be brought down into the test below 👍

File: __tests__/endpoint.test.js [hunk 2]
```diff
@@ -42,4 +42,40 @@ describe('Endpoint Tests', () => {
...
+        })
+        test('200: responds with ride object that matches paramentic ID given', () => {
+            return request(app).get('/api/articles/1').expect(200)
+            .then(({body}) => {
+                console.log(body.data, 'body in test')
```
Comment (nc-john-hutcheon) [inline __tests__/endpoint.test.js:52]: Remember to remove console logs before pushing code

File: __tests__/endpoint.test.js [hunk 3]
```diff
@@ -42,4 +42,40 @@ describe('Endpoint Tests', () => {
...
+            })
+        })
+        test('GET:400 sends an appropriate status and error message when given an invalid id', () => {
+            return request(app).get('/api/articles/not-an-id').expect(400)
+            .expect(400)
+            .then((response) => {
+                expect(response.body.msg).toBe('Bad Request')
+            });
+        });
+        test('DELETE:404 responds with an appropriate status and error message when given a non-existent id', () => {
```
Comment (nc-john-hutcheon) [inline __tests__/endpoint.test.js:72]: This test description says DELETE for some reason? Should be GET 🙂

File: app/app.js [hunk 4]
```diff
@@ -16,9 +19,18 @@ app.use((err, req, res, next) => {
     }
 })
 
+app.use((err, req, res, next) => {
+    if(err.code === '22P02'){
+        res.status(400).send({msg: 'Bad Request'})
+    }else{
+        next(err)
```
Comment (nc-john-hutcheon) [inline app/app.js:26]: Nice error handling 🙌

**Review by nc-john-hutcheon — COMMENTED** [review]
Great work Drew! I've just left a bit of feedback below 👍

File: controllers/controller.js [hunk 5]
```diff
@@ -21,4 +21,14 @@ exports.getEndpoints = (req, res, next) => {
     })
 }
 
+exports.getArticleID = (req, res, next) => {
+    const {article_id} = req.params;
+    selectArticleID(article_id)
+    .then ((data) => {
+        res.status(200).send({data});
```
Comment (nc-john-hutcheon) [inline controllers/controller.js:28]: I would send this back on a key of article rather than data, so we're being more descriptive

**Review by nc-john-hutcheon — COMMENTED** [review]
Just another comment i forgot to put!

**Pull request merged**